### ISSUE TITLE
Using DB::expr() in query builder method order_by() leads to an error

### ISSUE THREAD

**Issue by smgladkovskiy** [jonathangeiger/kohana-jelly#99]
Using DB:expr('RAND()')  in order_by() method returns an error:

```
ErrorException [ Warning ]: Illegal offset type in isset or empty

MODPATH/jelly/classes/jelly/meta/core.php [ 315 ]

310                 $this->fields += $field;
311                 return $this;
312             }
313         }
314         
315         if ( ! isset($this->field_cache[$field]))
316         {
317             $resolved_name = $field;
318             
319             if (isset($this->aliases[$field]))
320             {
```

The same query build by query builder works like expected. Looks like Jelly thinks it a field name...

**Comment by smgladkovskiy** [comment]
Something like this before line 315 helped in my problem:
    if($field instanceof Database_Expression)
    {
        return NULL;
    }

**Comment by banks** [comment]
There is actually a slightly bigger issue here - do we need to attempt to parse and alias the actual expression?

My feeling is we don't - we just need to return it without attempting to alias.

Thanks for the report.

**Comment by smgladkovskiy** [comment]
I can't imagine now an example, whenever required to make an alias of the expression in DB::expr (). So I agree that it should be returned as is.

**Comment by banks** [comment]
> I can't imagine now an example

I guess I meant what if the user wants to do

```
Jelly::update('post')->set(array('count' => DB::expr('count +1')))->execute();
```

In this scenario, the user would have to use the alias 'count' as the key but if the column was actually called view_count in the DB they would have to manually alias:

```
'count' => DB::expr('view_count + 1')
```

which isn't ideal.

Overall though I feel that use of DB::expr() represents enough of an edge case that complex parsing and aliasing support is wasted effort.

**Comment by smgladkovskiy** [comment]
What do you think that parse DB::expr() only if the value is set as an associative array, where the key - it is an alias, and value - an expression. Then do not lose functionality of aliasing and correctly can let SQL statements. If DB::expr() is introduced as a string, then handle it without trying to make from it an alias.

Somethind like this in Jelly_Bulde_Core, _column() method:

```
    $model = NULL;

    if(is_array($field))
    {
        foreach ($field as $alias => $value)
        {
            if($value instanceof Database_Expression)
            {
                return array($alias, $value);
            }
        }
    }
```

**Comment by banks** [comment]
NOt quite sure what you mean. The problem is that expressions are, by definition, arbitrary strings of SQL. That may contain columns, tables, aliases, functions - whatever you need in a query that doesn't fit into Query builder.

The only reliable solution to this would be to effectively parse the expressions for anythign that looks like a table or column name and attmept to alias it. This is probably not even possible to make it work 100% of the time so I feel it is best left well alone.

Using Expressions is by definition a pretty edge case so I don't think it is really a problem that developers will have to be responsible for manually aliasing (in other words, using actual table and column names).

**Comment by banks** [comment]
Clearly DB expressions should always WORK in any situation the work in Query BUilder - I just don't think we shold attempt to alter them for aliasing - the whole point of the Expression object is to provide the developer exact and ultimate control over the query so changing it is ridiculous - we would have to create a Database_Expression_Expression for when the developer wants to use an exact expression wihtout erroneous aliasing...

**Comment by smgladkovskiy** [comment]
I also think that the use of expressions imposes the responsibility for enforcing the code on the developer. And so I just don't see the point in spending time to solve the problem but to exclude Database_Expression objects from the aliasing routine...

**Comment by jonathangeiger** [comment]
Closed by 28b73ef712aeebcbeb9d2a48368d175011e0acac. Thanks smgladkovskiy!
